tests/units/classes/writers/file.php

```diff
@@ -203,13 +206,35 @@ public function testClear()
                     ->call('fopen')->withArguments($file->getFilename(), 'c')->once()
                     ->call('flock')->withArguments($resource, LOCK_EX)->once()
             ->if($adapter->ftruncate = true)
+            ->and($adapter->rewind = true)
             ->then
                 ->object($file->clear())->isIdenticalTo($file)
                 ->adapter($adapter)
                     ->call('fopen')->withArguments($file->getFilename(), 'c')->once()
                     ->call('ftruncate')->withArguments($resource, 0)
                         ->after($this->adapter($adapter)->call('flock')->withArguments($resource, LOCK_EX))
                             ->twice()
+                    ->call('rewind')->withArguments($resource)->once()
+        ;
+    }
+
+    public function testRewindWhenClearing()
+    {
+        $this
+            ->if($uri = stream_get_meta_data(tmpfile())['uri'])
```

Comment:
Maybe this would work better.
```suggestion
            ->if($uri = tempnam(sys_get_temp_dir(), 'foo'))
```

Reply:
Explanaition could be that the file is deleted in the middle of execution.

PHP documentation says: `The file is automatically removed when closed (for example, by calling fclose(), or when there are no remaining references to the file handle returned by tmpfile()), or when the script ends.`